pkg/net/libp2p/key_test.go

```diff
@@ -23,25 +25,37 @@ func TestOperatorPrivateKeyToNetworkKeyPair(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(networkPublicKey.Curve.Params(), DefaultCurve.Params()) {
-		t.Errorf("network public key uses the wrong curve")
+	secp256k1PublicKey, ok := networkPublicKey.(*libp2pcrypto.Secp256k1PublicKey)
+	if !ok {
+		t.Fatal("wrong type of public key")
 	}
 
-	if networkPublicKey.X.Cmp(operatorPublicKey.X) != 0 {
+	btcecPublicKey := (*btcec.PublicKey)(secp256k1PublicKey)
+
+	if btcecPublicKey.X().Cmp(operatorPublicKey.X) != 0 {
 		t.Errorf("network public key has a wrong X coordinate")
 	}
-	if networkPublicKey.Y.Cmp(operatorPublicKey.Y) != 0 {
+	if btcecPublicKey.Y().Cmp(operatorPublicKey.Y) != 0 {
 		t.Errorf("network public key has a wrong Y coordinate")
 	}
 
+	secp256k1PrivateKey, ok := networkPrivateKey.(*libp2pcrypto.Secp256k1PrivateKey)
+	if !ok {
+		t.Fatal("wrong type of private key")
+	}
+
+	btcecPrivateKey := (*btcec.PrivateKey)(secp256k1PrivateKey)
+
 	if !reflect.DeepEqual(
-		networkPrivateKey.PublicKey,
-		(ecdsa.PublicKey)(*networkPublicKey),
+		*btcecPrivateKey.PubKey(),
+		*btcecPublicKey,
 	) {
 		t.Errorf("network private key contains wrong network public key")
 	}
 
-	if networkPrivateKey.D.Cmp(operatorPrivateKey.D) != 0 {
+	d := extractD(btcecPrivateKey)
+
+	if d.Cmp(operatorPrivateKey.D) != 0 {
```

Comment:
Let's inline this. No need to extract `d`.

Reply:
Done in 64cc49ba0cbffe2de1aad6ee7c457097429ee97c.